docs/decisions/app/0005-use-next-auth.md

```diff
@@ -0,0 +1,123 @@
+# Use NextAuth over custom/provider specific solution for authentication
+
+- Status: [proposed]
+- Deciders: @sawyerh, @aligg, @lorenyu
+- Date: 2023-07-20
+- Technical Story: As a user, I want to be able to easily integrate authentication into my react application with common identity providers, such as login.gov
+
+## Context and Problem Statement
+
+A common requirement for government projects is working with an identity provider (Such as login.gov or AWS Cognito). It isn't uncommon for authentication providers to be changed during the lifecycle of a project. We want to ease the concern of being "locked in" to one provider. [NextAuth](https://next-auth.js.org/) fills this need by allowing us to easily swap identity providers and providing helpful functions for retrieving user data.
```

Comment:
not sure i agree with the part about

> It isn't uncommon for authentication providers to be changed during the lifecycle of a project.

i think the goal is less about making it easy to switch auth providers, and more about eliminating the dependency of the rest of the template on any particular auth provider, so that project teams can use the Next.js application template even if they use a non-default provider by simply implementing a new adapter rather than needing to refactor the parts of the application that rely on auth functionality.